lms/envs/common.py

```diff
@@ -912,6 +912,15 @@
 # e.g. COURSE_BLOCKS_API_EXTRA_FIELDS = [  ('course', 'other_course_settings'), ("problem", "weight")  ]
 COURSE_BLOCKS_API_EXTRA_FIELDS = []
 
+# .. toggle_name: REDIRECT_UNAUTHENTICATED_USER_TO_LOGIN_ON_ENROLL
+# .. toggle_implementation: DjangoSetting
+# .. toggle_default: False
+# .. toggle_description: If true, redirect unauthenticated users to the login page when enrolling from the course about
+#   page instead of showing an error message. After successful login redirects them back to the course about page.
+# .. toggle_use_cases: opt_in
+# .. toggle_creation_date: 2020-04-07
+# .. toggle_tickets: https://github.com/edx/edx-platform/pull/27197
+REDIRECT_UNAUTHENTICATED_USER_TO_LOGIN_ON_ENROLL = False
```

Comment:
Why make this a toggle? Seems like obviously better behavior. What good is that error page to the user?

But if we do leave this as a toggle:
- I'd like the name to mention that this is specifically for the Course About feature, which isn't always on. (our toggles in that space are confusing and overlapping - about vs info vs home?) So something like `COURSE_ABOUT_REDIRECT...` or similar.
- There's no easy way to set this variable? I'd expect a matching change in `production.py` where you pull an environment variable in to change this default, likely through our `edx/configuration` system. Or honestly, I'd expect this toggle to live in the `FEATURES` dictionary (which already has that configuration mechanism built in).

Reply:
@mikix, 

> Why make this a toggle? Seems like obviously better behavior. What good is that error page to the user?

Yes, it would be better off as the default behavior. We made this a toggle as we are not sure if this being used in any other way. If you think it will be good to remove this toggle, please let me know. :)

I've moved this to a feature flag named `COURSE_ABOUT_REDIRECT_UNAUTHENTICATED_USER_TO_LOGIN_ON_ENROLL`.

Reply:
Yeah, I vote for not bothering with a toggle here. And as a bonus, this PR would basically become a one-liner. Win-win.

Reply:
@mikix removed the toggle. I've also squashed all the commits, so this is ready to be merged. :) Thanks for the review.

Reply:
Love it, thanks! Will approve & merge. Thanks for your work!